modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PanelGroupTag.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.taglib.clay.servlet.taglib;
+
+import com.liferay.frontend.taglib.clay.internal.servlet.taglib.BaseContainerTag;
+import com.liferay.petra.string.StringPool;
+
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.jsp.JspException;
+
+/**
+ * @author Eduardo Allegrini
+ */
+public class PanelGroupTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+		setDynamicAttribute(StringPool.BLANK, "aria-orentation", "vertical");
+		setDynamicAttribute(StringPool.BLANK, "role", "tablist");
```

Comment:
Seems to me a bit odd to use `setDynamicAttribute`, here and in one more place. Is this applied to SSR? I think it ends up in `otherProps` of React component. We can simply manually set it, inline in SSR and the component.

Reply:
By default each Tag is inside an empty div, so this solution aimed to take advantage of it adding the missing classes and attributes required for this component. I can change it if necessary but it would be something like:

CURRENT:

```html
<div class="panel" aria-orentation="vertical" role="tablist">
  <button class="panel-header">..</button>
  <div class="panel-collapse">..</div>
</div>
```

NEW PROPOSAL:

```html
<div>
  <div class="panel" aria-orentation="vertical" role="tablist">
    <button class="panel-header">..</button>
    <div class="panel-collapse">..</div>
  </div>
</div>
```

Do we want that?

Reply:
By default each Tag is inside an empty div, so this solution aimed to take advantage of it adding the missing classes and attributes required for this component. I can change it if necessary but it would be something like:

CURRENT:

```html
<div class="panel" aria-orentation="vertical" role="tablist">
  <button class="panel-header">..</button>
  <div class="panel-collapse">..</div>
</div>
```

NEW PROPOSAL:

```html
<div>
  <div class="panel" aria-orentation="vertical" role="tablist">
    <button class="panel-header">..</button>
    <div class="panel-collapse">..</div>
  </div>
</div>
```

Do we want that?

---

EDIT:

Sorry I thought it was on the PanelTag.java file. But the concept is the same, adding the attributes inline will create an extra div. I've seen a similar pattern in common tags such Button, Alert, Checkbox, so I think it's an acceptable solution

Reply:
Ah, I see. Yeah, then it makes sense to add attributes that way :+1: